### PR TITLE
Frontier machine enviroment optimizations  (rebase to master)

### PR DESCRIPTION
**Pull request by mt5555**
Rebased version of Sarat's Frontier machine environment optimizations branch
https://github.com/E3SM-Project/scream/pull/2503

Also removes land changes (since the are already in our Frontier branch and wont be merged to master)
Removes some PIO changes which are already in master

### CODE REVIEW

**Comment by mt5555** [comment]
messed up this branch attempting to rebase onto  24d8f40  (the common parent of frontier branch and master), trying again...

**Comment by mt5555** [comment]
To merge this into the machines/frontier branch and resolve two minor conflicts, use:  

git merge --strategy-option theirs sarats/frontier-gb-rebase

**Comment by bartgol** [comment]
> To merge this into the machines/frontier branch and resolve two minor conflicts, use:
> 
> git merge --strategy-option theirs sarats/frontier-gb-rebase

Do we still need the `machines/frontier` branch after this PR? Can't it be reset to match `master` once this is merged?

**Review by sarats — APPROVED** [review]
Good to go for immediate use.

File: cime_config/machines/cmake_macros/crayclang-scream.cmake [hunk 1]
```diff
@@ -13,7 +13,7 @@ endif()
 string(APPEND CPPDEFS " -DFORTRANUNDERSCORE -DNO_R16 -DCPRCRAY")
 string(APPEND FC_AUTO_R8 " -s real64")
 # -em (default) generates MODULENAME.mod files
-string(APPEND FFLAGS " -f free -N 255 -h byteswapio -em")
+string(APPEND FFLAGS " -f free -em")
```
[the remark this answers is not in the document]
Comment (sarats) [reply cime_config/machines/cmake_macros/crayclang-scream.cmake:16]: These are obsolete Cray compiler flags.

**Pull request merged**